Formula/d/deck.rb

```diff
@@ -0,0 +1,31 @@
+class Deck < Formula
+  desc "Creates slide deck using Markdown and Google Slides"
+  homepage "https://github.com/k1LoW/deck"
+  url "https://github.com/k1LoW/deck/archive/refs/tags/v1.7.0.tar.gz"
+  sha256 "7136eba6db276237898099a38f62e6094b205be34748a539d6a6f8e95c19b6c5"
+  license "MIT"
+  head "https://github.com/k1LoW/deck.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = "-s -w"
+    system "go", "build", *std_go_args(ldflags:), "./cmd/deck"
+
+    (zsh_completion/"deck").write <<~EOS
+      #{opt_bin}/deck activate zsh | source
+    EOS
+    (fish_completion/"deck").write <<~EOS
+      #{opt_bin}/deck activate zsh | source
+    EOS
+    (bash_completion/"deck").write <<~EOS
+      #{opt_bin}/deck activate bash | source
+    EOS
```

Comment:
can you `Utils.safe_popen_read` to generate the completions?

Reply:
Leveraged that util in https://github.com/Homebrew/homebrew-core/pull/231816/commits/97fdee806c1580e115755a7f4b4bca1c5de361fb

Reply:
commit squashed

Reply:
Due to syntax issue found in https://github.com/Homebrew/homebrew-core/actions/runs/16639869396/job/47088090572?pr=231816, replaced with the DSL:

```ruby
generate_completions_from_executable(bin/"deck", "completion", shells: [:zsh, :bash, :fish])
```